src/vulkan/mod.rs

```diff
@@ -25,6 +25,9 @@ pub struct AllocationCreateDesc<'a> {
     pub location: MemoryLocation,
     /// If the resource is linear (buffer / linear texture) or a regular (tiled) texture.
     pub linear: bool,
+    /// True if the memory should be allocated in dedicated system memory.
+    /// If false, the memory will be placed in a potentially shared allocation block.
+    pub dedicated_allocation: bool,
```

Comment:
No need to write out "true"/"false" just like the doc comment above, that's inherent to the type (and we'll drop `if` as well). How about:

```suggestion
    /// This allocation should be placed in dedicated system memory according to the driver.
    /// It will otherwise be placed in a potentially shared allocation block.
    pub dedicated_allocation: bool,
```